components/prism-rescript.js

```diff
@@ -0,0 +1,48 @@
+(function (Prism) {
+	var operator = /\.{3}|:[:=]|\|>|->|=(?:==?|>)?|<=?|>=?|[|^?'#!~`]|[+\-*\/]\.?|\b(?:asr|land|lor|lsl|lsr|lxor|mod)\b/;
+	var className = /(\b[A-Z]\w*)|(@[a-z.]*)|#[A-Za-z]\w*|#\d/;
+	var number = /(?:\b0x(?:[\da-f]+(?:\.[\da-f]*)?|\.[\da-f]+)(?:p[+-]?\d+)?|(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:e[+-]?\d+)?)[ful]{0,4}/i;
+	var keyword = /\b(?:and|as|assert|begin|bool|class|constraint|do|done|downto|else|end|exception|external|float|for|fun|function|if|in|include|inherit|initializer|int|lazy|let|method|module|mutable|new|nonrec|object|of|open|or|private|rec|string|switch|then|to|try|type|val|virtual|when|while|with)\b/;
+
+	Prism.languages.rescript = {
+		'comment': {
+			pattern: /\/\/(?:[^\r\n\\]|\\(?:\r\n?|\n|(?![\r\n])))*|\/\*[\s\S]*?(?:\*\/|$)/,
+			greedy: true
+		},
+		'string': {
```

Comment:
2 things about strings:

1. Right now, `"${foo}"` is highlighted incorrectly. This can be fixed using an array:

```js
'string': [
  { pattern: /"..."/, greedy: true },
  { pattern: /`...`/, greedy: true, inside: ... }
]
```

2. The pattern for template strings doesn't actually handle interpolation. E.g. `` `foo${"`"}` `` is highlighted incorrectly.

To fix, I recommend just copying the [template string token from JS](https://github.com/PrismJS/prism/blob/master/components/prism-javascript.js#L123). Just replace the `rest: Prism.languages.javascript` with a `rest: null` and set the `rest` property after defining `Prism.languages.rescript`.

Reply:
About [string interpolation](https://rescript-lang.org/docs/manual/latest/primitive-types#string-interpolation): How do j-strings (that's what I'll call `` j`$foo` `` strings) compare to regular backtick strings? Do j-strings just *add* `$name` interpolations or do the `$name` interpolations *replace* `${name}` interpolations?

Reply:
My bad. This thread has already been resolved. Let's move the discussion about j-strings into its own thread.